sklearn/utils/estimator_checks.py

```diff
@@ -1720,6 +1721,47 @@ def transform(self, X):
                     shallow_params.items()))
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_set_params(name, estimator_orig):
+    if name in META_ESTIMATORS:
+        return
+    estimator = clone(estimator_orig)
+
+    # Trivial check to make sure set_params is working
+    params = estimator.get_params()
+    estimator.set_params(**params)
+
+    # Check that get_params() returns the same thing
+    # before and after set_params() with some fuzz
+    # values
+
+    estimator = clone(estimator_orig)
+    test_values = [-np.inf, np.inf, None,
+                   -100, 100, -0.5, 0.5, 0,
+                   "", "value",
+                   ('a', 'b'), {'key': 'value'}]
+
+    for param_name in params.keys():
+        for value in test_values:
+
+            try:
+                estimator.set_params(**{param_name: value})
+                get_value = estimator.get_params()[param_name]
```

Comment:
This should not be in the `try` block.

do we also want to check that all other params are unaffected?